tensorboard/webapp/reloader/reloader_component_test.ts

```diff
@@ -161,4 +182,158 @@ describe('reloader_component', () => {
 
     fixture.destroy();
   }));
+
+  it('does not reload if document is not visible', fakeAsync(() => {
+    store.setState(
+      createState(
+        createCoreState({
+          reloadPeriodInMs: 5,
+          reloadEnabled: true,
+        })
+      )
+    );
+    const fixture = TestBed.createComponent(ReloaderComponent);
+    fixture.detectChanges();
+
+    tick(5);
+    expect(dispatchSpy).toHaveBeenCalledTimes(1);
+
+    simulateVisibilityChange(false);
+    tick(5);
+    expect(dispatchSpy).toHaveBeenCalledTimes(1);
+
+    fixture.destroy();
+  }));
+
+  it('reloads when document becomes visible if missed reload', fakeAsync(() => {
+    store.setState(
+      createState(
+        createCoreState({
+          reloadPeriodInMs: 5,
+          reloadEnabled: true,
+        })
+      )
+    );
+    const fixture = TestBed.createComponent(ReloaderComponent);
+    fixture.detectChanges();
+
+    // Miss a reload because not visible.
+    simulateVisibilityChange(false);
+    tick(5);
+    expect(dispatchSpy).toHaveBeenCalledTimes(0);
+
+    // Dispatch a reload when next visible.
+    simulateVisibilityChange(true);
+    expect(dispatchSpy).toHaveBeenCalledTimes(1);
+
+    fixture.destroy();
+  }));
+
+  it('reloads when document becomes visible if missed reload, regardless of how long not visible', fakeAsync(() => {
```

Comment:
Is this longer than 80?

Reply:
Yes. But this seems to be how the formatter wants it (both the prettier vscode extension and prettier in `yarn fix-lint`.)  We could work around it by shortening or breaking up the string into multiple lines. Is it worth it?